guidance/hybridpath_guidance/hybridpath_guidance/hybridpath_guidance_node.py

```diff
@@ -0,0 +1,140 @@
+        self.guidance_publisher_ = self.create_publisher(HybridpathReference, "guidance/hybridpath/reference", 1)
+
+        # TODO: Add support for WaypointManager, add removal of "completed" waypoints
+
+        self.waypoints_ = waypoints
+        self.lambda_val_ = self.get_parameter('hybridpath_guidance.lambda_val').get_parameter_value().double_value
+        self.path_generator_order_ = self.get_parameter('hybridpath_guidance.path_generator_order').get_parameter_value().integer_value
+        self.time_to_max_speed_ = self.get_parameter('hybridpath_guidance.time_to_max_speed').get_parameter_value().double_value # Not used
+        self.dt_ = self.get_parameter('hybridpath_guidance.dt').get_parameter_value().double_value
+        
+        self.generator_ = HybridPathGenerator(waypoints, self.path_generator_order_, self.lambda_val_, 1)
+        self.path_ = self.generator_.Path
+
+        self.eta_d_ = np.zeros(3)
+        self.eta_d_prev_ = np.zeros(3)
+        self.eta_ = np.zeros(3)
+        self.eta_prev_ = np.zeros(3)
+        self.nu_ = np.zeros(3)
+        self.tau_ = np.zeros(3)
+        self.s_ = 0
+        self.u_desired_ = 1
+        
+        self.get_logger().info("hybridpath_guidance_node started")
+
+    def calculate_desired_states(self, dt: float) -> tuple[float, float, float, float, np.ndarray, np.ndarray, np.ndarray]:
+        """
+        Calculate the desired states for the guidance system
+
+        Args:
+            dt (float): Time step
+
+        Returns:
+            w_ref, v_ref, v_ref_t, v_ref_s, eta_d, eta_d_s, eta_d_ss
+        """
+        sig = HybridPathSignals(self.path_, self.s_)
+        v_ref, v_ref_s = sig.calc_vs(self.u_desired_)
+        self.s_ += v_ref * dt
+        pd = sig.pd  # Reference position
+        psi_d = sig.psi  # Reference heading
+        self.eta_d_ = np.array([pd[0], pd[1], psi_d])
+        psi0 = self.eta_d_prev_[2]
+        psi1 = self.eta_d_[2]
+        psi_vec = np.array([psi0, psi1])
+        psi_vec = np.unwrap(psi_vec, period=2 * np.pi)
+        self.eta_d_[2] = psi_vec[1]
+
+        # Variables needed for tau
+        v_ref_t = 0  # Constant speed
+        eta_d_s = np.array([sig.pd_der[0][0], sig.pd_der[0][1], sig.psi_der])
+        eta_d_ss = np.array([sig.pd_der[1][0], sig.pd_der[1][1], sig.psi_dder])
+        R, R_trsp = Rot(self.eta_prev_[2])
+        eta_error = self.eta_prev_ - self.eta_d_
+        my = 0.5
+        w_ref = my * (eta_d_s @ R @ (R_trsp @ eta_error)) / (np.linalg.norm(eta_d_s) ** 2)
+
+        return w_ref, v_ref, v_ref_t, v_ref_s, self.eta_d_, eta_d_s, eta_d_ss
+
+    def update_path(self, waypoints: np.ndarray) -> None:
+        """
+        Update the path generator with new waypoints
+
+        Args:
+            waypoints (np.ndarray): New waypoints
+        """
+        self.s_ = 0
+        self.generator_ = HybridPathGenerator(waypoints, self.path_generator_order_, self.lambda_val_, 1)
```

Comment:
Consider adding an update or method (or similar) to the `HybridPathGenerator` class so you avoid instantiating a new object here. Alternatively rename `update_path()` to signal that you are effectively resetting `self.generator_`